tests/test_meta.py

```diff
@@ -254,6 +255,75 @@ def test_commit_files(data_archive, cli_runner):
         assert r.exit_code == 44, r.stderr
 
 
+def test_commit_files_add_and_delete(data_working_copy, cli_runner):
```

Comment:
This test doesn't actually test your new functionality - if I revert your new functionality (or disable KART_ATTACHMENTS) the test still passes.

This test tests that the output of git ls-tree contains the expected attachment. git ls-tree is used for looking inside the Git object database, so, this is testing that the Git ODB contains the expected attachment. It is the job of `kart commit-files` to make sure that the attachment is saved to the ODB, and it was already working.

Your new code doesn't change the ODB - it changes what's on the filesystem. You can use the pathlib.Path library in this test to check what's on the file system and make sure the file you're concerned with doesn't exist before it is created, does exist once it is created and has the right contents, and doesn't exist after it is deleted.